habitat_baselines/common/obs_transformers.py

```diff
@@ -217,113 +218,496 @@ def from_config(cls, config: Config):
+        unproj_pts /= torch.norm(unproj_pts, dim=-1, keepdim=True)
+        # All points in image are valid
+        valid_mask = torch.full(unproj_pts.shape[:2], True, dtype=torch.bool)
+
+        # Rotate unproj_pts points according to camera rotation
+        if with_rotation:
+            unproj_pts = self.camcoord2worldcoord(unproj_pts)
+
+        return unproj_pts, valid_mask
+
+
+class EquirecCamera(CameraProjection):
+    """This is the equirectanglar camera class."""
+
+    def __init__(
+        self, img_h: int, img_w: int, R: Optional[torch.Tensor] = None
+    ):
+        """Args:
+        img_h: (int) the height of equirectanglar camera image
+        img_w: (int) the width of equirectanglar camera image
+        R: (torch.Tensor) 3x3 rotation matrix of camera
+        """
+        super(EquirecCamera, self).__init__(img_h, img_w, R)
+
+    def projection(
+        self, world_pts: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        # Rotate world points according to camera rotation
+        world_pts = self.worldcoord2camcoord(world_pts)
+
+        x, y, z = world_pts[..., 0], world_pts[..., 1], world_pts[..., 2]
+        # x,y,z to theta, phi
+        theta = torch.atan2(x, z)
+        c = torch.sqrt(x * x + z * z)
+        phi = torch.atan2(y, c)
+
+        # For grid_sample, -1 <= proj_pts <= 1
+        mapx = theta / np.pi
+        mapy = phi / (np.pi / 2)
+        proj_pts = torch.stack([mapx, mapy], dim=-1)
+
+        # All points in image are valid
+        valid_mask = torch.full(proj_pts.shape[:2], True, dtype=torch.bool)
+        return proj_pts, valid_mask
+
+    def unprojection(
+        self, with_rotation: bool = True
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        theta_map, phi_map = self.get_theta_phi_map(self.img_h, self.img_w)
+        unproj_pts = self.angle2sphere(theta_map, phi_map)
+        # All points in image are valid
+        valid_mask = torch.full(unproj_pts.shape[:2], True, dtype=torch.bool)
+        # Rotate unproj_pts points according to camera rotation
+        if with_rotation:
+            unproj_pts = self.camcoord2worldcoord(unproj_pts)
+        return unproj_pts, valid_mask
+
+    # Get theta and phi map
+    def get_theta_phi_map(self, img_h: int, img_w: int) -> torch.Tensor:
+        phi, theta = torch.meshgrid(torch.arange(img_h), torch.arange(img_w))
+        theta_map = (theta + 0.5) * 2 * np.pi / img_w - np.pi
+        phi_map = (phi + 0.5) * np.pi / img_h - np.pi / 2
+        return theta_map, phi_map
+
+    # Project on unit sphere
+    def angle2sphere(
+        self, theta_map: torch.Tensor, phi_map: torch.Tensor
+    ) -> torch.Tensor:
+        sin_theta = torch.sin(theta_map)
+        cos_theta = torch.cos(theta_map)
+        sin_phi = torch.sin(phi_map)
+        cos_phi = torch.cos(phi_map)
+        return torch.stack(
+            [cos_phi * sin_theta, sin_phi, cos_phi * cos_theta], dim=-1
+        )
+
+
+class FisheyeCamera(CameraProjection):
```

Comment:
Just for the sake of naming,  does this represent the projection for a single fisheye camera, or a dual fisheye camera. There is some overloading of names here and one might want to construct multiple fisheye cameras (say stereo fisheye cameras even!)

Reply:
Hmmm, I'm not sure if I understand correctly. But, maybe I should rename it as `FisheyeProjection`? If someones want `DualFishEye`, they can just use `FisheyeProjection` to define it. 
So, maybe `EquirecProjection` and `PerspectiveProjection` are better naming? What do you think?

Speaking of naming, you use `Equirec` for the abbreviation of equirectangular. I prefer to use `Equirect` since Unity uses `Equirect` for equirectangular [here](https://docs.unity3d.com/ScriptReference/RenderTexture.ConvertToEquirect.html), so it's easy to understand for many people.

Reply:
Sounds good. We are probably going to do a bigger refactor of ObsTransformers at some point anyway so more complete naming is better. Rename it to EquirectProjection and PerspectiveProjection. :)

Agree on your first point. Just wanted to make sure there wasn't term overloading. Leave Fisheye Projection as it is.